Normalize modal overlay dialogs (pre-PR)

luixxiul opened the pull request:
TODO: add modal dialog to `about:styles`.

Normalize modal overlay dialogs.

Follow up to 8042

----

Test Plan 1:
1. npm run test -- --grep='Payments'
2. npm run test -- --grep='Sync Panel'

Test Plan 2:
1. Open about:preferences#payments
2. Click "Add funds"
3. Make sure the 3 icons inside the dialog body is aligned

Test plan 3:
1. Open about:preferences#payments
2. Click "Add funds"
3. Click "Fund with credit/debit"
4. Make sure the coinbase widget is displayed at the center

Test Plan 4:
1. Add your country code to `coinbaseCountries.js`
2. Open about:preferences#payments
3. Click "Add funds"
4. Click "Display QR code"
5. Make sure the 3 icons appear on the footer

Test Plan 5:
1. Open about:preferences#payments
2. Enable payments
3. Click "Add funds"
4. Click "Display QR Code"
5. Make sure both the QR image and the label is aligned at the center

Test Plan 6:
1. Enable coinbase widget (via VPN)
2. Open about:preferences#payments
3. Click "Add funds"
4. Make sure "Done" button is aligned with the other buttons
5. Disable the coinbase widget
6. Make sure "Done button" is aligned to the right side

Test Plan 7:
1. Open about:preferences#payments
2. Click "Add funds"
3. Click "Display QR code"
4. Hover on the QR code image
5. Make sure its title is displayed

Test Plan 8:
1. Open about:preferences#sync
2. Enable Sync
3. Click "Sync a new device"
4. Click "Show secret code words"
5. Make sure the code is selectable

luixxiul commented on js/components/modalOverlay.js:
```diff
@@ -44,47 +50,153 @@ class ModalOverlay extends ImmutableComponent {
+    cursor: 'default',
+    width: '700px',
+    margin: '0 auto',
+    marginTop: '50px',
```
I'll replace `50px` with a variable

luixxiul commented on js/components/modalOverlay.js:
```diff
@@ -44,47 +50,153 @@ class ModalOverlay extends ImmutableComponent {
+
+  dialog__header: {
+    padding: '25px',
+    paddingLeft: '50px'
```
ditto

luixxiul replied:
Addressed https://github.com/luixxiul/browser-laptop/pull/1/commits/f69f09f3a020d485046b94e121b7d8a58b938051

luixxiul commented on js/components/modalOverlay.js:
```diff
@@ -44,47 +50,153 @@ class ModalOverlay extends ImmutableComponent {
+  },
+  dialog__body: {
+    background: '#fff',
+    padding: `${globalStyles.spacing.dialogInsideMargin} 50px`
```
ditto

luixxiul replied:
Addressed https://github.com/luixxiul/browser-laptop/pull/1/commits/f69f09f3a020d485046b94e121b7d8a58b938051

luixxiul commented on js/components/modalOverlay.js:
```diff
@@ -44,47 +50,153 @@ class ModalOverlay extends ImmutableComponent {
+
+  dialog__footer: {
+    padding: '20px',
+    paddingLeft: '50px',
```
ditto

luixxiul replied:
Addressed https://github.com/luixxiul/browser-laptop/pull/1/commits/f69f09f3a020d485046b94e121b7d8a58b938051

luixxiul commented:
@cezaraugusto right now I'm rebasing on the master branch

luixxiul commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -631,7 +615,9 @@ const styles = StyleSheet.create({
 
   syncOverlayBody__wrapper: {
```
This is no longer required. Will remove.

luixxiul replied:
Addressed with https://github.com/luixxiul/browser-laptop/pull/1/commits/7c5cd6143fb2a8fcffb7626eb60c5bc54d90cbae

luixxiul commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -249,9 +249,9 @@ class SyncTab extends ImmutableComponent {
-  get overlayContent () {
+  get newOverlayContent () {
     return <div className='syncOverlay'>
-      <div className={css(styles.syncOverlayBody__wrapper)}>
```
^

luixxiul commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -307,18 +307,15 @@ class SyncTab extends ImmutableComponent {
   }
 
   get startOverlayContent () {
-    return <div className={css(styles.syncOverlayBody__wrapper)}>
```
^

luixxiul commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -328,7 +325,7 @@ class SyncTab extends ImmutableComponent {
   }
 
   get addOverlayContent () {
-    return <div className={css(styles.syncOverlayBody__wrapper)}>
```
^

luixxiul commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -363,7 +357,7 @@ class SyncTab extends ImmutableComponent {
   }
 
   get resetOverlayContent () {
-    return <div className={css(styles.syncOverlayBody__wrapper)}>
```
^

luixxiul commented on app/renderer/components/styles/commonStyles.js:
```diff
@@ -63,20 +63,8 @@ const styles = StyleSheet.create({
-    padding: '25px 0 !important'
-  },
-
   modalOverlay__footer: {
```
This block becomes unnecessary too. See: https://github.com/luixxiul/browser-laptop/pull/1/files#diff-6e336f58ebfb0aaf5268c68f492d5c9cL94. WIll remove.

luixxiul replied:
`modalOverlay__footerButton` just under this is also no longer necessary.

luixxiul replied:
Addressed with https://github.com/luixxiul/browser-laptop/pull/1/commits/7c5cd6143fb2a8fcffb7626eb60c5bc54d90cbae

luixxiul commented:
TODO:
- Address my comments above
- Wrap function's content with `<section>` instead of `<div>` to make the hierarchy clear

cezaraugusto commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -249,9 +249,9 @@ class SyncTab extends ImmutableComponent {
 
-  get overlayContent () {
+  get newOverlayContent () {
     return <div className='syncOverlay'>
```
can you tell if removing this class would break something?

luixxiul replied:
I don't think so, going to remove it.

luixxiul replied:
Addressed https://github.com/luixxiul/browser-laptop/pull/1/commits/04e91768eb11730948f1569cce9b2974a87d6039

cezaraugusto commented on js/components/modalOverlay.js:
```diff
@@ -5,6 +5,12 @@
+const {StyleSheet, css} = require('aphrodite/no-important')
+const globalStyles = require('../../app/renderer/components/styles/global')
+
+const closeButton = require('../../img/toolbar/stoploading_btn.svg')
```
just ensure that when adding new images, running `npm run unittest` wouldn't break tests. If it does, you may want to mock the image. You can see how it is done by going to `publisherToggleTest.js`.

luixxiul replied:
yeah it breaks the test:
````
  1) Preferences component "before all" hook:
     SyntaxError: /Users/Suguru/browser-laptop/img/toolbar/stoploading_btn.svg: Unexpected token, expected } (1:88)
````

I'll look into it.

luixxiul replied:
Fixed with https://github.com/luixxiul/browser-laptop/pull/1/commits/26f25982ae454a97ab9dde287f014671c59fb723

cezaraugusto commented on js/components/modalOverlay.js:
```diff
@@ -44,47 +50,153 @@ class ModalOverlay extends ImmutableComponent {
 
-    let customTitleClassesStr = (this.props.customTitleClasses ? this.props.customTitleClasses : '')
     let customDialogClassesStr = (this.props.customDialogClasses ? this.props.customDialogClasses : '')
+    let customDialogHeaderClassesStr = (this.props.customDialogHeaderClasses ? this.props.customDialogHeaderClasses : '')
```
I'm ok with `let`, but if you assign a value to a variable that you won't change by yourself, it's preferable to use `const`. Consider both scenarios:

```js
// good use case for let
let luixxiul = 'Suguru'
// Let is writeable: I'm going to change (write) the value that I added previously:
luixxiul = 'awesome'
```

and

```js
const isSuguru = 'philosopher'
// Const is read-only, which means that once declared, I can not change
// but value may vary while I'm declaring it, I just can't change it after
const luixxiul = isSuguru ? 'Suguru here' : 'Nope, just Cezar breaking stuff'
```

also `()` is totally optional in such cases, if you really need it, either standard or compiler would complain. I'm ok if you prefer this way however.

luixxiul replied:
Addressed with https://github.com/luixxiul/browser-laptop/pull/1/commits/a7d5d3c97db35c85018ffd8ca5e7956dfea7915e

cezaraugusto commented on app/renderer/components/preferences/payment/bitcoinDashboard.js:
```diff
@@ -71,186 +111,241 @@ class BitcoinDashboard extends ImmutableComponent {
-        </a>
-      </div>
-    </div>
+    return <iframe className={css(styles.modalOverlay__coinbaseOverlay__bodyWrapper__body__iframe)}
```
can we have className less verbose? `styles.modalOverlay__coinbaseOverlay__iframe` would be good I think

luixxiul replied:
I'm fine with it.

luixxiul replied:
@cezaraugusto the component is inside `<body>` (you could see the hierachy more easily when you look at this line: https://github.com/luixxiul/browser-laptop/pull/1/files#diff-799b11467a9bfc86f7dbbf11fcaee159R637).

Despite its length and strictness, I belive the classname seems to make it easier to understand styling of the iframe, such as `height: 100%` and `width: 100%` on this line: https://github.com/luixxiul/browser-laptop/pull/1/files#diff-799b11467a9bfc86f7dbbf11fcaee159R639.

cezaraugusto commented on app/renderer/components/preferences/payment/bitcoinDashboard.js:
```diff
@@ -71,186 +111,241 @@ class BitcoinDashboard extends ImmutableComponent {
+    return <div className={css(styles.modalOverlay__qrcodeOverlay__content)}>
+      <img
+        src={this.props.ledgerData.get('paymentIMG')}
+        title='Brave wallet QR code'
```
not sure if we should translate this (makes sense in Portuguese as-is), but just fyi if you didn't know: we can translate titles going to .properties file and adding like 
```
qrcodeOverlayContentImg.title = some title here
```

l20n would then apply the string to title instead of plain text. Just a note

luixxiul replied:
oh I translated it already, maybe lost during rebase. Will fix soon.

luixxiul replied:
Addressed with https://github.com/luixxiul/browser-laptop/pull/1/commits/a7d5d3c97db35c85018ffd8ca5e7956dfea7915e

luixxiul replied:
Fixed with https://github.com/luixxiul/browser-laptop/pull/1/commits/7beb3e69416480606046a1cc298151e967a47949

cezaraugusto commented:
Left some comments, and would like to ask some polish regarding alignment:

#### 1. Align second row:
<img width="649" alt="screen shot 2017-05-01 at 1 04 48 pm" src="https://cloud.githubusercontent.com/assets/4672033/25588600/58a927ae-2e7f-11e7-8643-a735b5285ef6.png">
I was able to align that by removing subTitle margin and applying `justifyContent: space-between` to parent. It may have some caveats with other more verbose languages, so any other method to align that would be ok

#### 2. Align footer:

<img width="336" alt="screen shot 2017-05-01 at 1 04 58 pm" src="https://cloud.githubusercontent.com/assets/4672033/25588599/58a770f8-2e7f-11e7-9e57-d7e0913e9566.png">
I think setting `margin: 20px 30px` would do the trick:

#### 3. Align bitcoin footer:

<img width="463" alt="screen shot 2017-05-01 at 2 09 00 pm" src="https://cloud.githubusercontent.com/assets/4672033/25588601/58abd990-2e7f-11e7-86e5-fada5929a0f1.png">

BitCoin stuff looked weird. Maybe using flexbox and `justifyContent: space-around/space-between` would do the trick. Again, I'm ok with any other method as long as its aesthetically pleasant.

luixxiul commented:
@cezaraugusto uh, those stuff has been addressed with the latest upstream. I think they are reverted.... I'll fix them soon!

luixxiul commented:
@cezaraugusto By modifying the code I got this panel:

<img width="729" alt="screenshot 2017-05-02 9 57 34" src="https://cloud.githubusercontent.com/assets/3362943/25600269/de8ca3fc-2f1d-11e7-9d08-280843b6eebf.png">

I'm not quite sure how to reproduce those issues.

luixxiul commented:
@cezaraugusto concering ^, I'm wondering you have not removed the first 3 commits (which I removed yesterday). would you mind resetting all of the commits of this PR and pull again for verification? Thanks!

luixxiul commented:
Replaced `<div>`s with `<section>`s on files under `app/renderer/components/preferences/payment/`.

luixxiul commented:
@cezaraugusto the most work is done. When you have time, could you please review the commits again? there are many but simple and easy to review I think.

cezaraugusto commented:
LGTM. 
<img width="765" alt="screen shot 2017-05-02 at 4 26 51 pm" src="https://cloud.githubusercontent.com/assets/4672033/25635349/3566fe3a-2f54-11e7-8a23-e1c4a1515d36.png">

If we could align "done" with "display QR code" would be great, but OK to accept as-is. Great job!!

cezaraugusto commented:
I think you addressed already but if not, just ensure tests pass and the ones who fail are the same as the last Travis build. That's the only blocker I could see for this PR, everything else is perfect

luixxiul commented:
Thanks for reviewing! Appreciated 😄 

OK, this should look good:

<img width="745" alt="screenshot 2017-05-03 10 01 05" src="https://cloud.githubusercontent.com/assets/3362943/25645285/8b28ea98-2fe7-11e7-8c79-0ca108a143e4.png">
<img width="736" alt="screenshot 2017-05-03 10 01 38" src="https://cloud.githubusercontent.com/assets/3362943/25645286/8b29583e-2fe7-11e7-8957-0a5e8d1b4f1f.png">

luixxiul commented:
Squashed and merged to create a PR to the upstream repo. The commits messages were also rewritten.

Here are the latest screenshots of Add funds dialog. I think this should look much more normalized. 

<img width="733" alt="screenshot 2017-05-04 2 24 35" src="https://cloud.githubusercontent.com/assets/3362943/25673118/494608ec-3071-11e7-92dd-013deaf09dbb.png">
<img width="728" alt="screenshot 2017-05-04 2 24 43" src="https://cloud.githubusercontent.com/assets/3362943/25673120/49645ad6-3071-11e7-8ca7-bdffcf34ea7f.png">
<img width="734" alt="screenshot 2017-05-04 2 27 10" src="https://cloud.githubusercontent.com/assets/3362943/25673119/496371e8-3071-11e7-92ba-27f4fc689145.png">
<img width="733" alt="screenshot 2017-05-04 2 27 20" src="https://cloud.githubusercontent.com/assets/3362943/25673121/49648f42-3071-11e7-84d3-827c5054b790.png">

I also believe that especially the code in `bitcoinDashboard.js` became much more simpler and easy to understand.

@cezaraugusto - thanks very much for reviewing. I appreciate your work. I'm creating a PR to the upstream repo, adding you as a reviewer along with srinam. I'm sure nothing special was added/deleted after you have reviewed, so it should not be hard to review it there.
